src/main/java/wolox/training/models/Book.java

```diff
@@ -122,43 +122,75 @@ public String getIsbn() {
     }
 
     private void setGenre(String genre) {
-        this.genre = checkNotNull(genre);
+        Preconditions
+            .checkNotNull(genre, "The genre field cannot be null.");
+        Preconditions
+            .checkArgument(! genre.isEmpty(), "The genre field cannot be empty.");
```

Comment:
In java it is considered a good practice to place this type of messages as static variables in another class.